### ISSUE TITLE
RxNetty tests where the request body is read but response body is not written hang

### ISSUE THREAD

**Issue by rstoyanchev** [spring-projects/spring-reactive#58]
RxNetty integration tests where the request body is read but either there is no response writing (e.g. no response content) hang.

The RxNetty HttpServerRequest has a `discardContent` method which consumes the content if not read. However if the application does read the content and does not unsubscribe, under some circumstances, RxNetty will hang because it considers there to be an active subscriber still and waits for it to unsubscribe.

The current workaround is to add `.concatWith(Observable.empty())` inside the `RxNettyServerHttpRequest.getBody` method but we need to figure out why this does not happen otherwise in our case.

**Comment by rstoyanchev** [comment]
I've tried using a basic RxNetty sample to demonstrate the issue but so far haven't been able to. 

The only thing I've found so far is that use of Reactor's RxJava1Converter instead of the "official" RxJava - Reactive Streams bridge makes a difference. See for example the [modified RxNetty sample](https://github.com/rstoyanchev/RxNetty/tree/hanging-issue) here. Alternating between Reactor's and the official bridge demonstrates the issue.

However that alone is not the only issue since switching to the RxJava - Reactive Streams bridge doesn't fix the hanging in our integration tests. So there is something else still that's preventing unsubscribe from reaching the request body Observable.
